observation_portal/common/state_changes.py

```diff
@@ -273,9 +284,9 @@ def aggregate_request_states(request_group):
     """Aggregate the state of the request group from all of its child request states"""
     request_states = [request.state for request in Request.objects.filter(request_group=request_group)]
     # Set the priority ordering - assume AND by default
-    state_priority = ['WINDOW_EXPIRED', 'PENDING', 'COMPLETED', 'CANCELED']
+    state_priority = ['WINDOW_EXPIRED', 'PENDING', 'COMPLETED', 'RETRY_LIMIT', 'CANCELED']
     if request_group.operator == 'MANY':
-        state_priority = ['PENDING', 'COMPLETED', 'WINDOW_EXPIRED', 'CANCELED']
+        state_priority = ['PENDING', 'COMPLETED', 'WINDOW_EXPIRED', 'RETRY_LIMIT', 'CANCELED']
```

Comment:
Aggregating Request states into RequestGroup states is a little strange since it's not clear how what RequestGroup state corresponds to having Requests in a variety of states. This describes the order in which we decide the state, based on if any of the Requests are in that state. In this case I put RETRY_LIMIT behind WINDOW_EXPIRED, but I would be open to putting it in front of that instead if you think RETRY_LIMIT is a more important top level state then WINDOW_EXPIRED...

Reply:
I think this is reasonable. I honestly can't think of a strong case that would require one to go before the other.